File: 205-isomorphic-strings/isomorphic-strings.ts
function isIsomorphic(s: string, t: string): boolean {
    if (s.length !== t.length) return false;
    let smp = new Map<string, string>();
    let tmp = new Map<string, string>();

    for (let i = 0; i < s.length; i++) {
        if (smp.has(s.charAt(i))) {
            if (smp.get(s.charAt(i)) !== t.charAt(i)) return false;
        }
        else if (tmp.has(t.charAt(i))) {
            if(tmp.get(t.charAt(i)) !== s.charAt(i)) return false;
        }
        else {
            smp.set(s.charAt(i), t.charAt(i));
            tmp.set(t.charAt(i), s.charAt(i));
        }
    }
    return true;
};

File: 205-isomorphic-strings/isomorphic-strings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        if(s.length() != t.length()) return false;
        Map<Character , Character> sTot = new HashMap<>();
        Map<Character , Character> tTos = new HashMap<>();

        for(int i = 0; i < s.length(); i ++){
            Character curSChar = s.charAt(i);
            Character curTChar = t.charAt(i);
            if(sTot.containsKey(curSChar)){
                if(curTChar != sTot.get(curSChar)) return false;
            }
            if(tTos.containsKey(curTChar)){
                if(curSChar != tTos.get(curTChar)) return false;
            }
            sTot.put(curSChar,curTChar);
            tTos.put(curTChar,curSChar);
        }

        return true;

    }
}